libs/components/router/src/lib/modules/href/href.directive.ts

```diff
@@ -55,6 +57,8 @@ export class SkyHrefDirective {
     private router: Router,
     private renderer: Renderer2,
     private element: ElementRef,
+    private applicationRef: ApplicationRef,
+    private changeDetectorRef: ChangeDetectorRef,
```

Comment:
Since it's possible that folks are mocking this directive in their unit tests, any new parameters need to be `@Optional()`.

The only way around this is to provide a test fixture/harness so they don't need to mock it out themselves.